docs/ref/models/fields.txt

```diff
@@ -51,6 +51,12 @@ for "no data;" the Django convention is to use the empty string, not
 and ``blank=True`` set. In this situation, ``null=True`` is required to avoid
 unique constraint violations when saving multiple objects with blank values.
 
+Additionally, be cautious of depending on :attr:`~Field.null` to enforce a
+"not null" constraint at the database level for string-based fields, as
+databases may accept the empty string as a value despite :attr:`~Field.null`
+being ``False``. To ensure that a string-based field is not empty, you should
+set :attr:`default=None <Field.default>` and :attr:`null=False <Field.null>`.
```

Comment:
I'm not sure about this because this does enforce a not null constraint (it won't save null values)

I wonder if this is more about `blank=False` and saying that `blank=False` does not mean you won't be able to save blank values. You need to call `full_clean()` to run the validation

Reply:
Hello Sarah! Thank you for the review.
Perhaps this may be about ``blank=False`` too. But I hadn't thought about that since I experienced the issue when I created model instances in my views or in the interactive shell using ``create()`` or ``get_or_create()``, which is why I thought it was about ``null=False`` behavior on string-based fields when no ``default`` is specified. Specifically, how ``null=False`` will not disallow saving when no value is provided for sting-based field, which happens because database accepts empty strings as value.
Please, let me know what you think.

Reply:
Alright, let's keep the note in the null docs :+1: